src/pages/en/guides/cms/contentful.md

```diff
@@ -0,0 +1,445 @@
+
+## Creating a blog with Astro and Contentful
+
+In this section we will use Astro to create a blog with Contentful as the CMS. 
+
+### Prerequisites
+
+1. **A Contentful space** - For this tutorial we recommend starting with an empty space. However, if you already have content, feel free to use it. Keep in mind that you will need to modify the code snippets to match your content model.
+2. **An Astro project integrated with the [Contentful SDK](https://github.com/contentful/contentful.js)** - See [integrating with Astro](#integrating-with-astro) for more details on how to set up an Astro project with Contentful.
+
+### Setting up a Contentful model
+
+Inside your Contentful space, in **Content model** section, create a new content model with the following fields:
+
+- **Name:** Blog Post
+- **API identifier:** `blogPost`
+- **Description:** This content type is for a blog post
+
+In this newly created content type, create 5 new fields with the following parameters:
+
+1. Text field
+    - **Name:** title
+    - **API identifier:** `title`
+2. Date and time field
+    - **Name:** date
+    - **API identifier:** `date`
+3. Text field
+    - **Name:** slug
+    - **API identifier:** `slug`
+4. Text field
+    - **Name:** description
+    - **API identifier:** `description`
+5. Rich text field
+    - **Name:** content
+    - **API identifier:** `content`
+
+Now that you have your Contentful model set up, you can create your first blog post entry. In the **Content** section of your Contentful space, create a new entry with the following parameters:
+
+- **Content type:** `blogPost`
+- **Title:** `Astro is amazing!`
+- **Slug:** `astro-is-amazing`
+- **Description:** `Astro is a new static site generator that is blazing fast and easy to use.`
+- **Date:** `2021-10-01`
+- **Content:** `This is my first blog post!`
+
+Feel free to add as many blog posts as you want. Now that you have some data in your Contentful space, switch to your favorite code editor to start hacking with Astro!
+
+### Creating astro components
+
+Create a new astro component `Card.astro` inside the `src/components` directory of your project. This component will be used to display your blog post cards on the homepage later on.
+
+:::tip
+Not familiar with `.astro` files? Check out [astro components](/en/core-concepts/astro-components/) for more information.
+:::
+
+```astro title="src/components/Card.astro"
+---
+interface Props {
+  title: string;
+  description: string;
+  url: string;
+  date: string;
+}
+
+const { url, title, description, date } = Astro.props;
+---
+<li>
+  <a href={`/posts/${url}/`}>
+    <h2>{title}</h2>
+  </a>
+  <time>{date}</time>
+  <p>{description}</p>
+</li>
+```
+
+Create a new interface called `blogPostFields` and add it to your `contentful.ts` file in `src/lib`. This interface will match the fields of your blog post content type in Contentful and it will be used to type your blog post entries.
```

Comment:
I think this should be in its own file and named `BlogPost` rather than `blogPostFields`

Reply:
I wanted to avoid creating more files, but yes I am strongly biased in having a types file.